runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -297,33 +297,54 @@ function LanguageTree:_get_injections()
 
     for pattern, match, metadata in self._injection_query:iter_matches(root_node, self._source, start_line, end_line+1) do
       local lang = nil
-      local injection_node = nil
+      local ranges = {}
       local combined = false
 
+      -- Directives can configure how injections are captured as well as actual node captures.
+      -- This allows more advanced processing for determining ranges and language resolution.
+      if metadata.content then
+        local content = metadata.content
+
+        -- Allow for captured nodes to be used
+        if type(content) == "number" then
+          content = {match[content]}
+        end
+
+        if content then
+          vim.list_extend(ranges, content)
+        end
+      end
+
+      if metadata.language then
+        lang = metadata.language
+      end
```

Comment:
I think this is great work allowing the directive's to have control over the language and content of injected nodes. I suggested a similar idea here https://github.com/neovim/neovim/pull/14152 but without the ability to modify the content.

Merging this would enable a functioning `downcase!` / `lowercase!` directive in my PR as an example.